Helpers/CommonMenuItemPartMigrator.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Etch.OrchardCore.Menu.Helpers
+{
+    public class CommonMenuItemPartMigrator : ICommonMenuItemPartMigrator
+    {
+        private readonly AutorouteOptions _autorouteOptions;
+        private readonly IContentManager _contentManager;
+        private readonly ILogger<CommonMenuItemPartMigrator> _logger;
+        private readonly ISession _session;
+        private readonly ShellSettings _settings;
+
+        public CommonMenuItemPartMigrator(IOptions<AutorouteOptions> autorouteOptions,
+            IContentManager contentManager,
+            ILogger<CommonMenuItemPartMigrator> logger,
+            ISession session,
+            ShellSettings settings)
+        {
+            _autorouteOptions = autorouteOptions.Value;
+            _contentManager = contentManager;
+            _logger = logger;
+            _session = session;
+            _settings = settings;
+        }
+
+        public async Task GetCommonMenuitemDataAsync()
+        {
+            foreach (var contentItem in await FetchMenuItemsAsync())
+            {
+                var menuItemsListPart = await MenuItemMigratorAsync(contentItem.As<MenuItemsListPart>());
+
+                contentItem.Apply(nameof(MenuItemsListPart), menuItemsListPart);
+                await _contentManager.UpdateAsync(contentItem);
+                await _contentManager.PublishAsync(contentItem);
+            };
+        }
+
+        private async Task<string> ContentItemIdToUrlAsync(string contentItemId)
+        {
+            var contentItem = await _session.Query<ContentItem>()
+                .With<ContentItemIndex>()
+                    .Where(x => x.ContentItemId == contentItemId)
+                .FirstOrDefaultAsync();
+
+            if (contentItem == null)
+            {
+                return string.Empty;
+            }
+
+            if (contentItem.As<AutoroutePart>() != null)
+            {
+                return contentItem.As<AutoroutePart>().Path;
+            }
+
+            return string.Empty;
+        }
+
+        private async Task<IEnumerable<ContentItem>> FetchMenuItemsAsync()
+        {
+            return await _session.Query<ContentItem>()
+                .With<ContentItemIndex>()
+                    .Where(x => x.Latest && x.ContentType == "Menu")
+                .ListAsync();
+        }
+
+        public async Task<MenuItemsListPart> MenuItemMigratorAsync(MenuItemsListPart menuItemsListPart)
+        {
+            foreach (var linkMenuItem in menuItemsListPart.MenuItems)
+            {
+                var linkDestinationPart = new LinkDestinationPart();
+
+                try
+                {
+                    if (linkMenuItem.As<CommonMenuItem>().ExternalUrl.Text != null)
+                    {
+                        linkDestinationPart = new LinkDestinationPart { DestinationUrl = linkMenuItem.As<CommonMenuItem>().ExternalUrl };

```

Comment:
You already instantiate `LinkDestinationPart` on line 85. So this can change to `linkDestinationPart.DestinationUrl = linkMenuItem.As<CommonMenuItem>().ExternalUrl`